src/wallet/spend.cpp

```diff
@@ -162,6 +162,7 @@ util::Result<PreSelectedInputs> FetchSelectedInputs(const CWallet& wallet, const
 {
     PreSelectedInputs result;
     const bool can_grind_r = wallet.CanGrindR();
+    std::map<COutPoint, CAmount> map_of_bump_fees = wallet.chain().CalculateIndividualBumpFees(coin_control.ListSelected(), coin_selection_params.m_effective_feerate);
```

Comment:
non blocker, just a suggestion: In 35a35fa989f14406f8218c20c4e4e732cee0bd6c: you could move `map_of_bump_fees` to right before `ApplyBumpFee`, this way we calculate the individual bump fees when we really sure we're going to use it: 
```diff
diff --git a/src/wallet/spend.cpp b/src/wallet/spend.cpp
index 15004c67b0..db0870f31b 100644
--- a/src/wallet/spend.cpp
+++ b/src/wallet/spend.cpp
@@ -162,8 +162,8 @@ util::Result<PreSelectedInputs> FetchSelectedInputs(const CWallet& wallet, const
 {
     PreSelectedInputs result;
     const bool can_grind_r = wallet.CanGrindR();
-    std::map<COutPoint, CAmount> map_of_bump_fees = wallet.chain().CalculateIndividualBumpFees(coin_control.ListSelected(), coin_selection_params.m_effective_feerate);
-    for (const COutPoint& outpoint : coin_control.ListSelected()) {
+    const auto list_selected{coin_control.ListSelected()};
+    for (const COutPoint& outpoint : list_selected) {
         int input_bytes = -1;
         CTxOut txout;
         if (auto ptr_wtx = wallet.GetWalletTx(outpoint.hash)) {
@@ -198,6 +198,7 @@ util::Result<PreSelectedInputs> FetchSelectedInputs(const CWallet& wallet, const
 
         /* Set some defaults for depth, spendable, solvable, safe, time, and from_me as these don't matter for preset inputs since no selection is being done. */
         COutput output(outpoint, txout, /*depth=*/ 0, input_bytes, /*spendable=*/ true, /*solvable=*/ true, /*safe=*/ true, /*time=*/ 0, /*from_me=*/ false, coin_selection_params.m_effective_feerate);
+        std::map<COutPoint, CAmount> map_of_bump_fees{wallet.chain().CalculateIndividualBumpFees(list_selected, coin_selection_params.m_effective_feerate)};
         output.ApplyBumpFee(map_of_bump_fees.at(output.outpoint));
         result.Insert(output, coin_selection_params.m_subtract_fee_outputs);
     }
```

Reply:
Maybe I’m misunderstanding what you propose, but `CalculateIndividualBumpFees(…)` gets the bump fees for all elements of `coin_control.ListSelected()` at once, and this would move it into the for-loop, so we’d get the individual bump fees for all `n` elements `n` times for n² effort.